### PR TITLE
Issue 2688/madenearme - replace cgi by standalone script, remove a lot of unused code

### PR DESCRIPTION
**Pull request by stephanegigandet**
Rearchicted madenear.me : #2688

### CODE REVIEW

File: scripts/generate_madenearme_page.pl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# We want products with emb codes
+$query_ref->{"emb_codes_tags"} = { '$exists' => 1 };
+
+$request_ref->{map_options} = $map_options{$cc} || "";
+
+my $map_html = search_and_map_products($request_ref, $query_ref, $graph_ref);
+
+
+$html =~ s/<HEADER>/$header/;
```
Comment (hangy) [inline scripts/generate_madenearme_page.pl]: > [Sat Nov 30 18:12:25 2019] generate_madenearme_page.pl: Use of uninitialized value $header in substitution (s///) at generate_madenearme_page.pl line 99.

File: scripts/generate_madenearme_page.pl [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+$html =~ s/<INITJS>/$initjs/;
```
Comment (hangy) [inline scripts/generate_madenearme_page.pl]: > [Sat Nov 30 18:12:25 2019] generate_madenearme_page.pl: Use of uninitialized value $initjs in substitution (s///) at generate_madenearme_page.pl line 100.

**Comment by stephanegigandet** [comment]
> Would it make sense to just set them to `''` at the top of this script?

Well in fact it looks like we don't need them, the generated files seem to work quite well without them :)

**Comment by hangy** [comment]
> Well in fact it looks like we don't need them, the generated files seem to work quite well without them :)

ACK. We could, however, either initialize the variables or remove the replace (and placeholders) to get rid of the warnings.

**Comment by stephanegigandet** [comment]
OK, restricting the fields returned by MongoDB results in a huge speed improvement.
Details here:
https://github.com/openfoodfacts/openfoodfacts-server/issues/2688#issuecomment-560012616

We probably need to do something similar for the graphs, and possibly for the lists of products (although for those, since we retrieve only 20 products, it's not as bad as for the maps and graphs where we can process hundreds of thousands of products...)

**Comment by hangy** [comment]
> and possibly for the lists of products (although for those, since we retrieve only 20 products, it's not as bad as for the maps and graphs where we can process hundreds of thousands of products...)

Since we also return the whole result set without a projection from the JSON API, that could be a bad breaking change.

**Comment by stephanegigandet** [comment]
> ACK. We could, however, either initialize the variables or remove the replace (and placeholders) to get rid of the warnings.

Sure, I initialized them. I wasn't getting the warnings, they only appeared when the map had no results I think.

**Comment by stephanegigandet** [comment]
> Since we also return the whole result set without a projection from the JSON API, that could be a bad breaking change.

Right, we'll need to be careful, maybe we can do it when it's not an API request. We'll probably need to use separate memcached keys too.

**Pull request merged**